src/modules/template-details/TemplateDetails.tsx

```diff
@@ -10,115 +10,114 @@ import ShareIcon from '@material-ui/icons/Share';
 
 import { Button } from 'ui';
 
+import { getTemplateDetails, Template } from 'core/api';
+
 import csx from './TemplateDetails.scss';
 
+
+
 interface TemplateDetailsProps extends RouteChildrenProps<{ id: string }> {}
 
+// TODO - REPLACE MISSING PROPERTIES WHEN ENDPOINT WILL BE FINISHED
+// TODO - CONNECT EDIT
+
 const TemplateDetails = ({ match }: TemplateDetailsProps) => {
-  const MOCKED_TECH = ['PWA', 'React', 'JavaScript', 'TypeScript', 'MVP'];
-  const MOCKED_PATTERNS = ['PWA', 'MVP'];
   const MOCKED_TECH_STACK = ['React', 'Angular', 'Vue'];
   const MOCKED_AUTHORS = [
     'https://dummyimage.com/64x64/000/fff.png',
     'https://dummyimage.com/64x64/000/fff.png',
     'https://dummyimage.com/64x64/000/fff.png'
   ];
 
+  const [template, setTemplate] = useState<Template>(null);
+
   const mapList = (list: string[]) => list.map((item) => <li key={item}>{item}</li>);
   const mapImages = (list: string[]) =>
-    list.map((item) => (
-      <li key={item}>
+    // to avoid key error
+    list.map((item, idx) => (
+      <li key={item + idx}>
         <img src={item} />
       </li>
     ));
 
   useEffect(() => {
-    console.log(match);
-    console.log(`callApi with id ${match.params.id}`);
+    const getData = async () => {
+      const template = await getTemplateDetails(match.params.id);
+      setTemplate(template);
+    };
+
+    getData();
   }, [match.params.id]);
-  
+
+  if (template === null) return <div>Loading...</div>;
```

Comment:
After TemplateProvider you will have loading flag - and add this instead of template === null. Also avoid this ifs before templateDetails. Because if you would like add spinner or something much better approach will be add this inside templateDetails div and easier to center or smth like that. Also instead of loading add CircularProgress which should be centered